CHANGELOG.md

```diff
@@ -0,0 +1,15 @@
+# [slack-hubot] Changelog
+
+### v4.0.0 (2016-06-15)
+
+  * Now uses the latest version of `node-slack-sdk` (v3.4.1 as of this writing), inheriting all the improvements therein.
+  * Better (and automatically enabled) reconnect logic. As in, it actually reconnects automatically at all.
+  * Now you can upload files!
+  * Significantly improved handling of messages with attachments, which is to say, we can deliver them.
+  * Message formatting of links, usernames and channel names is now working far better than it ever did, which is damning with faint praise, but hey.
+  * Long messages are now left for Slack to handle, bless their hearts.
+  * Slack usernames with `.` and `-` are now treated with the respect and dignity due to all usernames.
+  * Messages from bots are no longer filtered out, which is both cool and potentially terrifying, but we should never have silenced the robots in the first place.
+  * Remember how if you tried to hack on this adapter and used `npm link` to plug that into a live bot? And how that didn't work? Yeah? Well now it does. Stupid `instanceof`.
+  * Total refactoring of the functionality, exposing a slightly different interface. So watch out for that.
+  * You can now access the underlying Slack client directly, for when you really need low-level functionality therein.
```

Comment:
[`.rtm` or `.web`](https://github.com/slackhq/hubot-slack/blob/09ba54d7ba3ff3993759bdfe7203ea11774c0d60/src/client.coffee#L15) will return the clients.

[`.send`](https://github.com/slackhq/hubot-slack/blob/09ba54d7ba3ff3993759bdfe7203ea11774c0d60/src/client.coffee#L73) is just a shortcut to select the best method.